include/swift/ABI/TaskOptions.h

```diff
@@ -141,6 +141,19 @@ class AsyncLetWithBufferTaskOptionRecord : public TaskOptionRecord {
   }
 };
 
+class ResultTypeInfoTaskOptionRecord : public TaskOptionRecord {
```

Comment:
Should this be `#if` to only be for embedded, or is it useful outside of that context? Also, should this store a `ResultTypeInfo` rather than repeating its fields.

Reply:
I was fretting about this in the definition of `ResultTypeInfo`. Should we have `ResultTypeInfo` always have the five fields, so we have consistent layout between "embedded" and non-embedded? That makes our development easier, and maybe possibly allows some level of interoperability between embedded-ish and non-embedded code, but it wastes a couple of pointers worth of storage per task.

Reply:
I'm not sure there's any prospect of such interoperability. In any case, the embedded side doesn't have to be ABI stable so we could fix this up later if need be.

Reply:
Made the class ifdef'd to embedded only, for now, and I'll explore the idea of using ResultTypeInfo in desktop Swift separately.